zephyr/lib/alloc.c

```diff
@@ -335,10 +381,24 @@ EXPORT_SYMBOL(rzalloc);
 void *rballoc_align(uint32_t flags, uint32_t caps, size_t bytes,
 		    uint32_t align)
 {
+	struct k_heap *heap;
+
+	/* choose a heap */
+	if (caps & SOF_MEM_CAPS_L3) {
+#if CONFIG_L3_HEAP
+		heap = &l3_heap;
+		return (__sparse_force void *)l3_heap_alloc_aligned(heap, align, bytes);
+#else
+		k_panic();
```

Comment:
this is kind of theoretical, but maybe a small follow-up PR to convert this and line 301 above to `return NULL`

Reply:
@lyakh this isn't theoretical based on the find by @marc-hb in the fuzzer logs